test/chpldoc/functions/WhereClause.doc.chpl

```diff
@@ -0,0 +1,35 @@
+module M {
+
+    record MyRecord {
+        var x: int;
+    }
+
+    proc processArr(arr: [1..n] int) where n != 0 {
+        writeln("processArr 1");
+    }
+
+    proc processArr(arr: []) where arr.elemType == int {
+        writeln("processArr 2");
+    }
+
+    proc processDom(dom: domain) where dom.rank == 2 {
+        writeln("processDom 1");
+    }
+
+    proc processRec(r: MyRecord) where r.x > 0 {
+        writeln("processRec");
+    }
+
+    proc processRange(r: [1..n] int) where n > 0 {
+        writeln("processRange 1");
+    }
+
+    proc processRange(r: range) where r.low > 1 {
+        writeln("processRange 2");
+    }
+
+    operator + (a: int, b: int) where a > 0 {
```

Comment:
these all look like good samples for testing the `where` clause. Can we add just a couple that include more complex logic in the where clause? We have several examples in `modules` that have where clauses that call other functions and/or include more complex logic. Here are some examples:

```
  inline operator <<(x: ?t, y: _tuple) where isHomogeneousTuple(y) &&
                                             isCoercible(t, (y(0).type)) {
```

```
  inline operator :(x: c_string, type t:c_ptrConst(?eltType))
    where eltType == c_char || eltType == int(8) || eltType == uint(8)
  {
    return __primitive("cast", t, x);
  }
```